test/state/requests.cpp

```diff
@@ -63,33 +71,77 @@ std::optional<Requests> collect_deposit_requests(std::span<const TransactionRece
             //
             // In ABI a word with its size prepends every bytes array.
             // Skip over the first 5 words (offsets of the values) and the pubkey size.
-            // TODO: EIP requires to read these offsets and validate them.
-            //       This has not been implemented yet because there are no tests for it.
-            static constexpr auto PUBKEY_OFFSET = 32 * 5 + 32;
-            static constexpr auto PUBKEY_SIZE = 48;
-            // Pubkey size is 48 bytes, but is padded to the word boundary, so takes 64 bytes.
-            // Skip over the pubkey and withdrawal credentials size.
-            static constexpr auto WITHDRAWAL_CREDS_OFFSET = PUBKEY_OFFSET + 64 + 32;
-            static constexpr auto WITHDRAWAL_CREDS_SIZE = 32;
-            // Skip over withdrawal credentials and amount size.
-            static constexpr auto AMOUNT_OFFSET = WITHDRAWAL_CREDS_OFFSET + 32 + 32;
-            static constexpr auto AMOUNT_SIZE = 8;
-            // Pubkey size is 8 bytes, but is padded to the word boundary, so takes 32 bytes.
-            // Skip over amount and signature size.
-            static constexpr auto SIGNATURE_OFFSET = AMOUNT_OFFSET + 32 + 32;
-            static constexpr auto SIGNATURE_SIZE = 96;
-            // Skip over signature and index size.
-            static constexpr auto INDEX_OFFSET = SIGNATURE_OFFSET + 96 + 32;
-            static constexpr auto INDEX_SIZE = 8;
+            // Read and validate the ABI offsets and lengths for the dynamic fields
+            // according to EIP-6110. If any check fails, collection is considered failed.
+            auto read_word_as_size = [&](size_t pos) -> std::optional<size_t> {
```

Comment:
is there a standard function to accomplish this? I would expect so, but couldn't find any